src/packages/core/efuns_main.cc

```diff
@@ -351,10 +351,22 @@ void f_capitalize(void) {
 #endif
 
 #ifdef F_CHILDREN
+//TODO:fix this. I basically moved code from the old children into this function but it should probably
+//be some sort of helper in array.h which works for all types. In future this "fix" probably won't be 
+//if I make changes to array_t.
 void f_children(void) {
-  array_t *vec;
+  auto max_array_size { CONFIG_INT(__MAX_ARRAY_SIZE__ ) };
+  auto vec {allocate_empty_array(max_array_size)};
```

Comment:
vec here seems to be misleading, because it is a array_t ,not a std::vector.  I would name it as “res" or something

Reply:
Also , since you should already know the result size, why always allocate an huge array and resize it later, this will probably not very efficient. 

this code should be rewritten as follows:

```
const auto max_array_size = ...;
auto vec = ObjecTable::get()->children(sp->u.string); 
auto len = min(vec.size(), max_array_size);
auto res = allocate_empty_array(len);
....
```

Reply:
I actually just recycled the name from the original code. It can be changed.

Reply:
The allocating one large array was from the original code as well.

Reply:
Fixed.